apps/api/src/app/workflows-v2/util/template-parser/parser-types.ts

```diff
@@ -0,0 +1,10 @@
+export type InvalidVariable = {
+  context: string;
+  message: string;
+  variable: string;
+};
+
+export type TemplateParseResult = {
+  validVariables: string[];
```

Comment:
It's better DX for this parse function to return the same type of variables for valid and invalid. That is consider refactoring the types as follows:

```
export type Variable = { name: string, message? string, context?: string };
export type TemplateParseResult = { validVariables: Variable[], invalidVariables: Variable [] }
```

Reply:
Love it, my main problem is this part https://github.com/novuhq/novu/pull/7192/files#r1871222390
if you want we can sync about it